### ISSUE TITLE
pull data about covid infections from government website

### ISSUE THREAD

**Issue by n400peanuts** [covid19ABM/comma#45]

**Comment by jiqicn** [comment]
https://github.com/J535D165/CoronaWatchNL

**Comment by n400peanuts** [comment]
Let me have a think about this, but it's great that the info is in principle extractable and it covers a quite wide range of time. 
At the first glance, especially considering the fact that you pulled also age distribution, I was wondering whether we could use these data also to generate our population and add whether they are infected or not already from step 1

**Comment by n400peanuts** [comment]
Linking here also issue #6 for completeness

**Comment by jiqicn** [comment]
@n400peanuts I would go with road nr.1 as it better models reality. But indeed let's discuss that on Monday. BTW, can you point me to the likelihood information of getting infected given the number of people encountered? Not sure if I have seen it in the dataset repos.

**Comment by n400peanuts** [comment]
@jiqicn it's an info we have to derive from the literature, if @kristinathompson  could advise us on how to implement it it would be great. The likelihood of getting infected it's not going to be trivial to estimate it. My initial thought would be to have likelihood of becoming positive (or we can call it infection probability) to be a composite variable that is based on: (i) % of infected people at the population level for that period of time -- we can derive that from the data that you pulled out from the notebook (ii) how many people you have encountered, and the chances that a fraction of those or all of them might be positive, this can be derived by the matrix of effects of actions on number of contacts (iii) the agents' age, which according from the literature I found on the topic is a factor. Also, we need to know how becoming infected impact the agents: impact on mental health and impact on the actions (we need to probably add quarantine when positive?).

Let's discuss it today with Kristina and let's see if she has suggestions

**Comment by n400peanuts** [comment]
OK, this is my plan on how to implement positivity for covid-19 in the model.

--------------------
Positivity estimation
--------------------

1. We start at step 0 with _n_i_ being the number of new positives in the target population for that day _i_. We get this info by the test data available from RIVM in the github repository mentioned by Ji in his notebook --> we read that info in the dataframe from row i.
2. We compute the number of new positives in day i for our simulated population as _m = (n/N)*M_ where _N_ is the size of the real population (e.g., in groningen -- data available from the government statistics), _M_ is the size of the simulated population, _n_ is the number of new positives reported on RIVM for that day
3. We randomly extract _m_ agents from the non-positive ones, these become positive at day i
4. We track for each positive agent the number of day _k_ that they have been positive
5. Each day, we flip a coin for each positive agent, with a probability _P(k)_ of negativization, that depends only on the number of day passed since the first positive day (e.g., with a gamma distribution, based on literature), if the coin is head the agent becomes negative
6. If "negativisation" happens after a threshold (e.g., 20 days, again this threshold depends on literature about diagnoses of long-covid) the agent is marked as long-covid

NOTE: when an agent is marked as positive / long-covid: 
1. this selects the relative row in the actions matrix
2. it resets the probability of choosing actions --> it will activate the action "quarantine"

**Comment by Astrid-p** [comment]
Hi Eva, 

This is really nice plan. Just a small contribution if it is possible. What do you think if we stratify _m_  by age group? It would be more realistic because I know that, for some certain lockdowns, the government targeted young people due to the increase of infections mostly among this age group (closing down the discotheque, bars, and clubs). And also put the minimum number of positive days is 5 (according to RIVM). So that we can only choose a new covid-positive agent to replace the old ones (among _m_), as well as we can put a higher negative probability on the positive agents after this period.

**Comment by jiqicn** [comment]
@n400peanuts sounds like a nice plan! 

Just one tip: in (2), you may rarely get the number of new positives to be greater than 1 for 95% of the steps by calculating that way, since _n_ (typically around 100) is much smaller than _N_ (e.g. 238,179 for Groningen in 2023).

**Comment by n400peanuts** [comment]
Good catch, @jiqicn. Let me try it out in the notebook and then I see how I can solve this
